light/detector.go

```diff
@@ -119,7 +119,7 @@ func (c *Client) compareNewHeaderWithWitness(ctx context.Context, errc chan erro
 
 	// the witness hasn't been helpful in comparing headers, we mark the response and continue
 	// comparing with the rest of the witnesses
-	case provider.ErrNoResponse, provider.ErrLightBlockNotFound, context.DeadlineExceeded, context.Canceled:
```

Comment:
By removing `provider.ErrNoResponse, provider.ErrLightBlockNotFound,` here, they will both be wrapped with `errBadWitness` in the default case and the provider will be removed / blacklisted